### PR TITLE
[TT-303] Implement field level depth limit

### PR DESCRIPTION
**Pull request by furkansenharputlu**
https://tyktech.atlassian.net/browse/TT-303

### CODE REVIEW

File: gateway/session_manager.go [hunk 1]
```diff
@@ -270,18 +278,57 @@ func (l *SessionLimiter) ForwardMessage(r *http.Request, currentSession *user.Se
...
-func (l *SessionLimiter) DepthLimitExceeded(r *http.Request, apiLimit *user.APILimit, schema *graphql.Schema) sessionFailReason {
-	gqlRequest := ctxGetGraphQLRequest(r)
+func (l *SessionLimiter) DepthLimitEnabled(graphqlEnabled bool, accessDef *user.AccessDefinition) bool {
+	if !graphqlEnabled {
+		return false
+	}
+
+	// This is more precise way, however getting fields from gqlRequest is complex at this point.
+	// Getting this info on `gqlRequest.ValidateForSchema` step would work.
+	/*for _, fieldRight := range accessDef.FieldAccessRights {
```
Comment (pvormste) [inline gateway/session_manager.go:288]: Do we want to keep this commented code?

**Review by pvormste — APPROVED** [review]

**Review by pvormste — APPROVED** [review]

**Review by spetrunin — APPROVED** [review]

**Pull request merged**